reconcile/openshift_resources_base.py

```diff
@@ -817,3 +858,79 @@ def run(
         sys.exit(1)
 
     return ri
+
+
+def early_exit_desired_state(
```

Comment:
This function is a complicated beast. I think it is a good moment to add a comment to guide the new reader of this.

Reply:
added comments in [c22bed7](https://github.com/app-sre/qontract-reconcile/pull/2734/commits/c22bed7b258faa8f4afd65c148e1d8377a64bc8a)

Reply:
wunderbar! ❤️